client/my-sites/stats/site.jsx

```diff
@@ -438,16 +470,27 @@ const enableJetpackStatsModule = ( siteId, path ) =>
 export default connect(
 	( state ) => {
 		const siteId = getSelectedSiteId( state );
+		const canUserManageOptions = canCurrentUser( state, siteId, 'manage_options' );
 		const isJetpack = isJetpackSite( state, siteId );
 		const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
-		// Odyssey: if Stats is not enabled, the page will not be rendered.
+
+		// Odyssey Stats: This UX is not possible in Odyssey as this page would not be able to render in the first place.
 		const showEnableStatsModule =
 			! isOdysseyStats &&
 			siteId &&
 			isJetpack &&
-			isJetpackModuleActive( state, siteId, 'stats' ) === false;
+			isJetpackModuleActive( state, siteId, 'stats' ) === false &&
+			canUserManageOptions;
+
+		// Odyssey Stats: Access control is done in PHP, so skip capability check here.
+		// TODO: Fix incorrect view_stats permission on Calypso.
+		//       If the user's role is missing from the site's stats dashboard access allowlist (fetched via getJetpackSettings.role),
+		//       then it should be reflected in the user's view_stats capability.
+		const canUserViewStats =
+			! isOdysseyStats && ( canUserManageOptions || canCurrentUser( state, siteId, 'view_stats' ) );
```

Comment:
If I didn't get it wrong, we should all Odyssey access here:
```suggestion
			isOdysseyStats || ( canUserManageOptions || canCurrentUser( state, siteId, 'view_stats' ) );
```

Otherwise Odyssey will always show:

<img width="477" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/2b5c57fc-bcb9-467f-9939-f028f1a3735b">

Reply:
I got this completely backward. Thanks for the suggestion!